operations.py

```diff
@@ -38,12 +64,341 @@ def kleene_star(automaton1: Automaton) -> Automaton:
+                    found = True
+
+            if not found:
+                return False
+            
+    return True
+
+
+def has_repeated_transitions(automaton: Automaton) -> bool:
+    for s in sorted(automaton.states):
+        if len(s.transitions) != len(automaton.alphabet):
+            print(f"state {s.name} has {len(s.transitions)} and automaton has {len(automaton.alphabet)}")
+            return True
+
+        for alph in automaton.alphabet:
+            found = False
+            for trans in s.transitions:
+                if trans.symbol == alph:
+                    if not found:
+                        found = True
+                    else:
+                        return True
+                        
+    return False
+
+def has_epsilon(automaton: Automaton) -> Automaton:
+    for trans in automaton._transitions:
+        if trans.symbol == EPSILON_SYMBOL:
+            print("Theres an epsilon")
+            return True
+        
+    return False
+
+def get_afd_from_afn(automaton: Automaton) -> Automaton:
+
+    if has_epsilon(automaton):
+        automaton1 = remove_epsilon(automaton)
+    else:
+        automaton1 = automaton
+
+    if has_repeated_transitions(automaton1):
+        afd = remove_repeated_transitions(automaton1)
+    else:
+        afd = automaton1
+
+    return afd
+
+def take_target(elem):
+    return elem.target
+
+def clean_target(target: str,alphabet: set) -> str:
+
+    states = None
+    for alph in sorted(alphabet):
+        if alph.name in target:
+            if states == None:
+                states = alph.name
+            else:
+                states += alph.name
+    
+    return states
+
+def dfs(initial: State, states: set) -> str:
+    q1 = queue.Queue()
+    node_names = None
+
+#    for st in states:
+#        if initial == st.name:
+    q1.put(initial)
+#            break
+        
+#    print(f"Starting dfs")
+    while not q1.empty():
+
+        s = q1.get()
+
+#        print(f"Poping {s.name}")
+        if s.worked == False:
+            s.worked = True
+            if node_names == None:
+                node_names = s.name
+            else:
+                node_names += s.name
+
+            for trans in s.transitions:
+                if trans.symbol != EPSILON_SYMBOL:
+                    continue
+#                print(f"Adding {trans.target}")
+                for st in states:
+                    if trans.target == st.name:
+                        q1.put(st)
+                        break
+    
+    #Clean previous states
+    for s in states:
+        s.worked = False
+
+    print(f"Nodes reached from {st.name} are {node_names}")
+    return node_names
+
+def remove_epsilon(automaton: Automaton) -> Automaton:
+
+    cerradura = dict()
+    states = set()
+#    return automaton
+
+    for s in automaton.states:
+        s.worked = False
+
+    for s in sorted(automaton.states):
+        target = s.name
+        for trans in sorted(s.transitions, key=take_target):
```

Comment:
There are 2 ways to get this cleaner:

Use lambda:
```
for trans in sorted(s.transitions, key=lambda x: x.target):
```

Create a `__lt__` function in `class Transitions()`